client/lib/resize-image-url/index.js

```diff
@@ -39,20 +54,43 @@ const scaleByFactor = ( value ) => value * IMAGE_SCALE_FACTOR;
  * Changes the sizing parameters on a URL. Works for WordPress.com, Photon, and
  * Gravatar images
  *
- * @param   {String} imageUrl Original image url
- * @param   {Object} params   Resize parameters to add
- * @returns {String}          Resize image URL
+ * @param   {String}          imageUrl Original image url
+ * @param   {(Number|Object)} resize   Resize pixel width, or object of query
+ *                                     arguments (assuming Photon or Gravatar)
+ * @param   {?Number}         height   Pixel height if specifying resize width
+ * @returns {String}                   Resize image URL
  */
-export default function resizeImageUrl( imageUrl, params ) {
+export default function resizeImageUrl( imageUrl, resize, height ) {
+	if ( 'string' !== typeof imageUrl ) {
+		return imageUrl;
+	}
+
 	const parsedUrl = parse( imageUrl, true, true );
 	if ( ! REGEXP_VALID_PROTOCOL.test( parsedUrl.protocol ) ) {
 		return imageUrl;
 	}
 
 	parsedUrl.query = omit( parsedUrl.query, SIZE_PARAMS );
 
+	if ( 'number' === typeof resize ) {
```

Comment:
My personal preference here is to avoid having a function with multiple signatures. Though, it's not easy to find good names in these cases. But granted this is a personal preference.